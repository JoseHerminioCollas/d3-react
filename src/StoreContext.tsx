import React from 'react'
import defaultState, { StateInterface } from './state'
import reducer, { dispatchInterface } from './reducer'
import useAction, { AppActionsInterface } from './action'

const StoreContext = React.createContext<any>(defaultState)

const StoreProvider: React.FC = ({ children }) => {
  const [state, dispatch]: [StateInterface, dispatchInterface] = React
    .useReducer(reducer, defaultState)
  const actions: AppActionsInterface = useAction(state, dispatch)

  return (
    <StoreContext.Provider value={{ state, actions }}>
      {children}
    </StoreContext.Provider>
  )
}
const StoreConsumer = StoreContext.Consumer

export { StoreContext, StoreProvider, StoreConsumer }
