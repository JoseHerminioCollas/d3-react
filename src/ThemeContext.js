import React, { createContext, useState, useEffect } from 'react'
import jss from 'jss'
import preset from 'jss-preset-default'
import themeFactory, { menuOptions } from './theme-factory'

jss.setup(preset())

const ThemeContext = createContext({})
const themeNames = menuOptions
const ThemeProvider = ({ children }) => {
  const [themeName, setThemeName] = useState('redStyle')
  const sheet = jss.createStyleSheet(themeFactory(themeName))
  sheet.attach()
  const [cssSheet, setCssSheet] = useState(sheet)
  console.log('s', setCssSheet)
  useEffect(() => {
    const newSheet = jss.createStyleSheet(themeFactory(themeName))
    console.log('eee', newSheet)
    // newSheet.attach()
    // setCssSheet(newSheet)
  }, [themeName])


  return (
    <ThemeContext.Provider value={{ themeName, setThemeName, cssSheet }}>
      {children}
    </ThemeContext.Provider>
  )
}

export { ThemeContext, ThemeProvider, themeNames }
