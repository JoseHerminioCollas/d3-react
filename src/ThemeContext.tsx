import React, { createContext, useState } from 'react'

const defaultTheme = { a: 1 }
const ThemeContext = createContext<any>(defaultTheme)
const ThemeProvider: React.FC = ({ children }) => {
  const [theme, setTheme] = useState(defaultTheme)
  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  )
}

export { ThemeContext, ThemeProvider }
