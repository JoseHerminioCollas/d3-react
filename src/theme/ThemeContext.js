import React, { createContext, useState, useEffect } from 'react'
import jss from 'jss'
import preset from 'jss-preset-default'
import themeFactory, { menuOptions } from './theme-factory'

jss.setup(preset())

const ThemeContext = createContext({})
const themeNames = menuOptions
const ThemeProvider = ({ children }) => {
  const [themeName, setThemeName] = useState('gray')
  const sheet = jss.createStyleSheet(themeFactory(themeName))
  sheet.attach()
  const [cssSheet, setCssSheet] = useState(sheet)
  useEffect(() => {
    const newSheet = jss.createStyleSheet(themeFactory(themeName))
    newSheet.attach()
    setCssSheet(newSheet)
  }, [themeName])


  return (
    <ThemeContext.Provider value={{ themeName, setThemeName, cssSheet }}>
      {children}
    </ThemeContext.Provider>
  )
}

export { ThemeContext, ThemeProvider, themeNames }
