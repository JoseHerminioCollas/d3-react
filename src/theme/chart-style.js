const chartStyle = {
  chartFrame: {
    overflow: 'visible',
    color: 'red',
    fill: 'blue',
    stroke: 'green',
    padding: '10px',
    margin: '33px',
    minWidth: '300px',
    maxWidth: '700px',
  },
  chartXAxis: {
    fontSize: '11px',
    color: 'black',
    fill: 'transparent',
    stroke: 'black',
    strokeWidth: '1px',
  },
  chartYAxis: {
    fontSize: '11px',
    color: 'black',
    fill: 'transparent',
    stroke: 'black',
    strokeWidth: '1px',
  },
  chartBars: {
    fontSize: '11px',
    color: 'orange',
    fill: 'black',
    stroke: 'green',
    strokeWidth: 0,
    width: '500px',
    height: '300px',
  },
  chartText: {
    fill: 'black',
    stroke: 'green',
    strokeWidth: '0',
    fontSize: '16px',
  },
  chartPies: {
    fill: 'lightgray',
    stroke: 'black',
  },
  chartPiesText: {
    fontSize: '10px',
    fill: 'white',
    stroke: 'transparent',
    strokeWidth: '0',
  },
  chartIconForce: {
    fill: 'green',
    stroke: 'black',
  },
  titleArea: {
    position: 'fixed',
    display: 'flex',
    justifyContent: 'center',
    bottom: 0,
    left: 0,
  },
}
export default chartStyle
