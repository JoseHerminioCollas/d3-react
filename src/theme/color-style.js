const colorStyle = {
  backgroundA: {
    fill: 'white',
  },
  backgroundB: {
    fill: '#111',
  },
  foregroundA: {
    fill: '#fff',
  },
  foregroundB: {
    fill: '#eee',
  },
  accent: {
    fill: 'red',
    stroke: 'orange',
  },
}
export default colorStyle
// const steps = 9
// let i = 0
// let j = 1
// const cSCopy = JSON.parse(JSON.stringify(colorStyle))
// Object.keys(cSCopy).forEach(k => {
//   cSCopy[k].stroke = interpolator(i)
//   cSCopy[k].fill = interpolator(j)
//   i += 1 / steps
//   j -= 1 / steps
// })
// Object.assign(defaultStyle, cSCopy)
