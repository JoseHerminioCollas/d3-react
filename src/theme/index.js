import ColorScheme from 'color-scheme'

const scheme = new ColorScheme()
const schemes = [
  'mono',
  'contrast',
  'triade',
  'tetrade',
  'analogic']
const variations = [
  'pastel',
  'soft',
  'light',
  'hard',
  'pale']

export { variations, scheme, schemes }
