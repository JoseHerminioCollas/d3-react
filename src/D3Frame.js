import React, { useContext } from 'react'
import Info from './component/Info'
import Control from './component/Control'
import Chart from './component/Chart'
import TitleArea from './component/TitleArea'
import { StoreContext } from './StoreContext'
import { ThemeContext } from './theme/ThemeContext'

function D3Frame() {
  const { cssSheet } = useContext(ThemeContext)
  return (
    <StoreContext.Consumer>
      {({ state, actions }) => {
        return (
          <div className={cssSheet.classes.mainContainer}>
            <Chart />
            <TitleArea />
            <Control />
            {
              state.isInfoVisible && <Info onClick={actions.hideInfo} />
            }
          </div>
        )
      }}
    </StoreContext.Consumer>
  )
}

export default D3Frame
