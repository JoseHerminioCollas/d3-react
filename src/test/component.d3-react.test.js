import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
/* eslint-disable-next-line */
import jss from 'jss'
import D3Frame from '../D3Frame'
import { StoreProvider } from '../StoreContext'
import { ThemeProvider } from '../theme/ThemeContext'

jest.mock('jss', () => {
  return {
    setup: () => 1,
    createStyleSheet: () => {
      return {
        attach: () => { },
        classes: { mainContainer: {} },
      }
    },
  }
})

Enzyme.configure({ adapter: new Adapter() })

describe('D3 Framework', () => {
  it('renders without crashing', () => {
    Enzyme.mount(
      <ThemeProvider>
        <StoreProvider>
          <D3Frame />
        </StoreProvider>
      </ThemeProvider>,
    )
  })
})
