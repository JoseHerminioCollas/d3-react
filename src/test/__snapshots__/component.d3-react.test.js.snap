// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`D3 Framework shows the Info element 1`] = `
<section
  data-id="container"
>
  <svg
    className="main"
    data-chart-id="pie-chart"
    data-component-type="container"
    data-id="chart-frame"
    height={300}
    width={700}
  >
    <rect
      data-id="chart-frame-background"
      fill="blue"
      height={300}
      width="100%"
    />
    <g
      style={
        Object {
          "transform": "translate(55px, 50px)",
        }
      }
    >
      <g
        data-id="pie-chart-labels"
        style={
          Object {
            "transform": "translate(295px, 137px)",
          }
        }
      >
        <text
          x={86.01455316868521}
          y={68.56746052750843}
        >
          A
        </text>
        <text
          x={-16.82164378919939}
          y={-108.7061741587353}
        >
          B
        </text>
        <text
          x={-99.38983278937803}
          y={-47.134500507584406}
        >
          C
        </text>
        <text
          x={-109.9455386984042}
          y={-3.460999901430637}
        >
          D
        </text>
        <text
          x={101.74087629807639}
          y={-41.81858546268058}
        >
          E
        </text>
        <text
          x={-60.224156690584806}
          y={-92.04917680733429}
        >
          F
        </text>
        <text
          x={-49.03374632609426}
          y={-98.46670361715293}
        >
          G
        </text>
        <text
          x={-92.17870272236986}
          y={60.02571752524881}
        >
          H
        </text>
        <text
          x={8.319017900891643}
          y={109.68497591358921}
        >
          I
        </text>
        <text
          x={44.5743592680264}
          y={-100.56404176366874}
        >
          J
        </text>
        <text
          x={-5.112561130942436}
          y={-132.9016994574651}
        >
          K
        </text>
        <text
          x={-106.38153560978314}
          y={-27.981581108015362}
        >
          L
        </text>
        <text
          x={-81.62217550924842}
          y={-73.74157894388652}
        >
          M
        </text>
        <text
          x={-32.328466079432815}
          y={105.14214322026616}
        >
          N
        </text>
        <text
          x={49.853677324447034}
          y={98.05412208178663}
        >
          O
        </text>
        <text
          x={-26.874826829653454}
          y={-106.6665068466955}
        >
          P
        </text>
        <text
          x={-1.066850944443508}
          y={-160.99646526884476}
        >
          Q
        </text>
        <text
          x={-106.6408082012705}
          y={26.976619991760224}
        >
          R
        </text>
        <text
          x={-66.99647425484933}
          y={87.24375299939423}
        >
          S
        </text>
        <text
          x={107.73356381863178}
          y={22.214392337779415}
        >
          T
        </text>
        <text
          x={-91.16415313138117}
          y={-61.55564298939688}
        >
          U
        </text>
        <text
          x={-10.2416802410208}
          y={-118.55845809490222}
        >
          V
        </text>
        <text
          x={-71.37688187160394}
          y={-83.69791355993944}
        >
          W
        </text>
        <text
          x={-1.9561363043111109}
          y={-146.98698422227378}
        >
          X
        </text>
        <text
          x={-38.05409069338942}
          y={-103.20797537738686}
        >
          Y
        </text>
        <text
          x={-0.3531378335594624}
          y={-174.99964369583873}
        >
          Z
        </text>
      </g>
      <g
        data-id="pie-chart-pies"
        style={
          Object {
            "transform": "translate(300px, 130px)",
          }
        }
      >
        <path
          d="M90.03171098099911,43.52345365241418A100,100,0,0,1,62.49597797696558,78.06569500556971L0,0Z"
        />
        <path
          d="M-19.299370917372695,-98.11999939968237A100,100,0,0,1,-11.260125337537627,-99.36402556953368L0,0Z"
        />
        <path
          d="M-93.34210431670589,-35.87828816609283A100,100,0,0,1,-86.8469202722446,-49.57431229201661L0,0Z"
        />
        <path
          d="M-99.64313477540706,8.440716328017714A100,100,0,0,1,-98.91496100006754,-14.691170489621241L0,0Z"
        />
        <path
          d="M74.09219383501788,-67.15911563380016A100,100,0,0,1,99.9049509740916,-4.3589873668492185L0,0Z"
        />
        <path
          d="M-59.86036937943413,-80.1045328165498A100,100,0,0,1,-49.42503889098314,-86.9319592015779L0,0Z"
        />
        <path
          d="M-49.42503889098314,-86.9319592015779A100,100,0,0,1,-39.59267434772854,-91.82820992480862L0,0Z"
        />
        <path
          d="M-73.61786659878693,67.67872425986775A100,100,0,0,1,-91.67094084948094,39.95545774698328L0,0Z"
        />
        <path
          d="M26.25442429680912,96.49199554804076A100,100,0,0,1,-11.401012770334507,99.34795875009547L0,0Z"
        />
        <path
          d="M6.123233995736766e-15,-100A100,100,0,0,1,74.09219383501788,-67.15911563380016L0,0Z"
        />
        <path
          d="M-5.946661582798292,-99.8230294872865A100,100,0,0,1,-1.7396967178556835,-99.98486613147954L0,0Z"
        />
        <path
          d="M-98.91496100006754,-14.691170489621241A100,100,0,0,1,-93.34210431670589,-35.87828816609283L0,0Z"
        />
        <path
          d="M-78.43752589782741,-62.02865894913134A100,100,0,0,1,-69.64712842616294,-71.75846641330602L0,0Z"
        />
        <path
          d="M-11.401012770334507,99.34795875009547A100,100,0,0,1,-46.38536991029988,88.59118160564654L0,0Z"
        />
        <path
          d="M62.49597797696558,78.06569500556971A100,100,0,0,1,26.25442429680912,96.49199554804076L0,0Z"
        />
        <path
          d="M-29.496363018418503,-95.55084807936387A100,100,0,0,1,-19.299370917372695,-98.11999939968237L0,0Z"
        />
        <path
          d="M-0.9216909554674894,-99.99575233869992A100,100,0,0,1,-0.40358527378553416,-99.99918559131761L0,0Z"
        />
        <path
          d="M-91.67094084948094,39.95545774698328A100,100,0,0,1,-99.64313477540706,8.440716328017714L0,0Z"
        />
        <path
          d="M-46.38536991029988,88.59118160564654A100,100,0,0,1,-73.61786659878693,67.67872425986775L0,0Z"
        />
        <path
          d="M99.9049509740916,-4.3589873668492185A100,100,0,0,1,90.03171098099911,43.52345365241418L0,0Z"
        />
        <path
          d="M-86.8469202722446,-49.57431229201661A100,100,0,0,1,-78.43752589782741,-62.02865894913134L0,0Z"
        />
        <path
          d="M-11.260125337537627,-99.36402556953368A100,100,0,0,1,-5.946661582798292,-99.8230294872865L0,0Z"
        />
        <path
          d="M-69.64712842616294,-71.75846641330602A100,100,0,0,1,-59.86036937943413,-80.1045328165498L0,0Z"
        />
        <path
          d="M-1.7396967178556835,-99.98486613147954A100,100,0,0,1,-0.9216909554674894,-99.99575233869992L0,0Z"
        />
        <path
          d="M-39.59267434772854,-91.82820992480862A100,100,0,0,1,-29.496363018418503,-95.55084807936387L0,0Z"
        />
        <path
          d="M-0.40358527378553416,-99.99918559131761A100,100,0,0,1,-1.8369701987210297e-14,-100L0,0Z"
        />
      </g>
    </g>
  </svg>
  <section
    data-id="control"
  >
    <form>
      <label
        data-id="control.info"
      >
        <button
          name="info"
          onClick={[Function]}
          value="show"
        >
          ?
        </button>
      </label>
      <label
        data-id="control.type"
      >
        Type
        <select
          name="chartType"
          onChange={[Function]}
          value="pie"
        >
          <option
            value="line"
          >
            Line
          </option>
          <option
            value="pie"
          >
            Pie
          </option>
          <option
            value="bar"
          >
            Bar
          </option>
        </select>
      </label>
      <label
        data-id="control.color"
      >
        Color
        <select
          name="color"
          onChange={[Function]}
          value="blue"
        >
          <option
            value="red"
          >
            Red
          </option>
          <option
            value="blue"
          >
            Blue
          </option>
          <option
            value="green"
          >
            Green
          </option>
          <option
            value="white"
          >
            White
          </option>
          <option
            value="gray"
          >
            Gray
          </option>
        </select>
      </label>
    </form>
  </section>
  <section
    data-id="info-min"
  >
    <a
      href="https://github.com/JoseHerminioCollas/d3-react"
      target="new"
    >
      D3 Framework
    </a>
    <a
      href="http://goatstone.com"
      target="new"
    >
      Goatstone Â© 2018
    </a>
  </section>
</section>
`;
